modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -598,7 +622,23 @@ public void stop() {
 
         busyLock.block();
 
-        Map<UUID, TableImpl> tables = tablesByIdVv.latest();
+        Map<UUID, TableImpl> tables;
+
+        try {
+            // Waiting for tablesByIdVv generally can be unbounded, because we don't have time limits for receiving storage revision update,
+            // where tablesByIdVv is completed, so we add timeout here.
+            tables = tablesByIdVv.waitForLatest(TABLES_COMPLETE_TIMEOUT, TimeUnit.SECONDS);
+        } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            throw new IgniteException(TABLE_NOT_COMPLETED_ERR);
+        }
+
+        cleanUpTablesResources(tables);
```

Comment:
shouldnt we save some context of the tables that are being created, we already have the futures for them (beforeTablesVvComplete) - they can be enriched